src/components/textArea/TextArea.styles.tsx

```diff
@@ -0,0 +1,112 @@
+import styled from "styled-components";
+
+export const TextAreaContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: 4px;
+`;
+
+export const Heading = styled.div<{ disabled?: boolean }>`
+  color: ${(props) =>
+    props.disabled
+      ? "var(--neutral-200, #9ca3af)"
+      : "var(--primary-300, #115873)"};
+  font-size: 16px;
+  font-style: normal;
+  font-weight: 700;
+  line-height: 150%;
+  letter-spacing: 0.019px;
+`;
+
+export const InputContainer = styled.div<{
+  isFocused: boolean;
+  disabled?: boolean;
+}>`
+  display: flex;
+  padding: 12px 8px;
+  width: 320px;
+  align-items: center;
+  gap: 8px;
+  align-self: stretch;
+  border-radius: 6px;
+  border: 1px solid
+    ${(props) =>
+      props.isFocused
+        ? "var(--neutral-200, #414141)"
+        : "var(--neutral-200, #9ca3af)"};
+  background: ${(props) =>
+    props.disabled ? "var(--neutral-50, #F9FAFB)" : "var(--system-50, #fff)"};
+  box-shadow: 0px 1px 2px 0px rgba(0, 0, 0, 0.05);
+  flex-shrink: 0;
+  pointer-events: ${(props) => (props.disabled ? "none" : "auto")};
```

Comment:
Same as above, destructure and move it at the end. Also can we move the dynamic properties in a single method that are using same prop instead of making multiple methods?